x-pack/plugins/ml/public/application/datavisualizer/index_based/data_loader/data_loader.ts

```diff
@@ -90,10 +97,24 @@ export class DataLoader {
     return stats;
   }
 
+  async loadFieldHistograms(
+    fields: FieldHistogramRequestConfig[],
+    query: string | SavedSearchQuery,
+    samplerShardSize = DEFAULT_SAMPLER_SHARD_SIZE
+  ): Promise<any[]> {
+    const stats = await ml.getVisualizerFieldHistograms({
```

Comment:
can you please use `mlApiServices` as an injected dependency instead of the static import from the cache? it's a part of `MlGlobalServices` you can retrieve from the context

Reply:
The rest of the file uses the same approach and I didn't plan to refactor it in this PR.